Description: can be dropped once gnu-efi << 3.0.8 is gone
--- a/shim.h
+++ b/shim.h
@@ -28,6 +28,21 @@
 #include <efilib.h>
 #undef uefi_call_wrapper
 
+// The following were introduced in gnu-efi 3.0.6
+#ifndef gST
+#define gST ST
+#endif
+#ifndef gBS
+#define gBS BS
+#endif
+#ifndef gRT
+#define gRT RT
+#endif
+// The following was introduced in gnu-efi 3.0.5
+#ifndef EFI_LOADED_IMAGE_PROTOCOL_GUID
+#define EFI_LOADED_IMAGE_PROTOCOL_GUID LOADED_IMAGE_PROTOCOL
+#endif
+
 #include <stddef.h>
 
 #define min(a, b) ({(a) < (b) ? (a) : (b);})
--- a/lib/console.c
+++ b/lib/console.c
@@ -99,7 +99,7 @@ console_print(const CHAR16 *fmt, ...)
 		setup_console(1);
 
 	va_start(args, fmt);
-	ret = VPrint(fmt, args);
+	ret = VPrint((CHAR16 *)fmt, args);
 	va_end(args);
 
 	return ret;
@@ -118,7 +118,7 @@ console_print_at(UINTN col, UINTN row, c
 	co->SetCursorPosition(co, col, row);
 
 	va_start(args, fmt);
-	ret = VPrint(fmt, args);
+	ret = VPrint((CHAR16 *)fmt, args);
 	va_end(args);
 
 	return ret;
--- a/MokManager.c
+++ b/MokManager.c
@@ -1914,8 +1914,7 @@ static BOOLEAN check_der_suffix(CHAR16 *
 	if (!file_name || StrLen(file_name) <= 4)
 		return FALSE;
 
-	suffix[0] = '\0';
-	StrnCat(suffix, file_name + StrLen(file_name) - 4, 4);
+	StrCpy(suffix, file_name + StrLen(file_name) - 4);
 
 	StrLwr(suffix);
 	for (i = 0; der_suffix[i] != NULL; i++) {
